packages/svelte/src/internal/client/reactivity/effects.js

```diff
@@ -96,7 +96,30 @@ function create_effect(type, fn, sync) {
 		effect.component_function = dev_current_component_function;
 	}
 
-	if (current_reaction !== null && !is_root) {
+	if (sync) {
+		var previously_flushing_effect = is_flushing_effect;
+
+		try {
+			set_is_flushing_effect(true);
+			execute_effect(effect);
```

Comment:
Is it safe to execute the effect before linking it to its parent? If we have error boundaries in the future, I imagine we have to walk up the effect tree, but it's not constructed yet then.

Reply:
The parent doesn't need to know about the child as long as the child knows about the parent, surely?

Reply:
oh right, the parent is set - yeah that is probably enough